### PR TITLE
update multisig address

### PR DESCRIPTION
**Pull request by thelostone-mc**
##### Description

Untitled

### CODE REVIEW

**Comment by nutrina** [comment]
@gdixon @thelostone-mc 
Here is where the polygon net id is kind of "set by brute force" ... this is why the fix does not work ...

image

**Comment by nutrina** [comment]
@gdixon @thelostone-mc as far as I have tested, when you are on polygon network, and attempt to to checkout on zkzync the correct address used for the gitcoin grant. 
This however is a bit messy ....

File: app/assets/v2/js/cart-ethereum-zksync.js [hunk 1]
```diff
@@ -111,6 +119,13 @@ Vue.component('grantsCartEthereumZksync', {
       appCart.$refs.cart.handleError(e);
     },
 
+    // Wait for correct chainId
+    waitForCorrectChain() {
+      return new Promise((resolve, _reject) => {
+        this.stopWaitingForNetwork = resolve;
```
Comment (gdixon) [inline app/assets/v2/js/cart-ethereum-zksync.js:125]: So we create a new `promise` every time `onChangeHandler` is called and then we resolve it when `networkId` is called - but to what end? I'm struggling to understand what this is trying to do? Couldn't we just rely on `networkId` as a global?

File: app/assets/v2/js/cart.js [hunk 2]
```diff
@@ -375,6 +374,12 @@ Vue.component('grants-cart', {
       return isPolygon ? 'MATIC' : 'ETH';
     },
 
+    gitcoinAddress() {
+      let isPolygon = this.networkId === '80001' || this.networkId === '137';
```
Comment (gdixon) [inline app/assets/v2/js/cart.js:378]:
```suggestion
      let isPolygon = networkId === '80001' || networkId === '137';
```

File: app/assets/v2/js/cart-ethereum-zksync.js (hunk 1 above)
Comment (nutrina) [reply app/assets/v2/js/cart-ethereum-zksync.js:125]: @gdixon  We ar enot waiting for the `networkId`, we are actually waiting for the change of networkId to be propagated to the `cart.js` which in turn will update the address in the Gitcoin grant.
The way it is build is misleading I have to admit ... 
Maybe the cleanest way to handle is would be to update the address of the Gitcoin grant in `cart-ethereum-zksync.js`, in the checkout function. Then this logic would not span across multiple files at least?

File: app/assets/v2/js/cart.js (hunk 2 above)
Comment (nutrina) [reply app/assets/v2/js/cart.js:378]: @gdixon : I do not think this will work. You want to use the data attribute `networkId` of the vue component?

**Comment by nutrina** [comment]
@thelostone-mc @gdixon I would suggest an alternative approach to solve this topic, therefore I have created this PR: #9917
Nevertheless, I think some functionality which is used for both chains (ethereum and polygon) could be moved to a utils.js file, for clarity.

**Pull request closed without merging**